### PR TITLE
Feature/job builder/update job detail modal content

### PR DESCRIPTION
**Pull request by xander-miller**
Updates Job Preview Modal Content and adds localization to Job Preview Modal Content

### CODE REVIEW

File: resources/assets/js/components/JobPreview.tsx [hunk 1]
```diff
@@ -11,36 +17,112 @@ export interface JobPreviewProps {
...
   securityLevel,
-  startDate,
   classification,
+  telework,
   level,
-}: JobPreviewProps): React.ReactElement => {
+  intl,
+}: JobPreviewProps & InjectedIntlProps): React.ReactElement => {
   return (
     <div
```
Comment (Khristophor) [inline resources/assets/js/components/JobPreview.tsx]: Looks like the majority of the markup has changed for this component, see
https://github.com/GCTC-NTGC/TalentCloud/blob/feature/job-builder/step-03-ui/resources/views/manager/builder-02.html.twig#L455-L521

File: resources/assets/js/components/JobPreview.tsx [hunk 2]
```diff
@@ -68,67 +150,183 @@ const JobPreview: React.FunctionComponent<JobPreviewProps> = ({
...
+        />
       </h4>
       <div data-c-grid="gutter">
         <div data-c-grid-item="tp(1of2)">
           <p data-c-colour="c1" data-c-margin="bottom(quarter)">
-            Average Annual Salary
+            <FormattedMessage
+              id="jobPreview.lengthOfTheTerm"
+              defaultMessage="Length of the Term"
+              description="Datum Label"
```
Comment (Khristophor) [inline resources/assets/js/components/JobPreview.tsx]: I know it's kind of annoying to have to fill out all of these descriptions, but I believe the spirit of the property is to add context to the line being translated. When a translator receives a file with hundreds of items needing translation, it's not very clear what the context is surrounding the lines.

Having `Datum Label` in this case for most of the translations in this file isn't super helpful to that end.

File: resources/assets/js/components/JobPreview.tsx [hunk 3]
```diff
@@ -68,67 +150,183 @@ const JobPreview: React.FunctionComponent<JobPreviewProps> = ({
...
+        </div>
+        <div data-c-grid-item="tp(1of2)">
+          <p data-c-colour="c1" data-c-margin="bottom(quarter)">
+            <FormattedMessage
+              id="jobPreview.travel"
+              defaultMessage="Travel"
+              description="Datum Label"
+            />
+          </p>
+          <p>Not yet implemented.</p>
```
Comment (Khristophor) [inline resources/assets/js/components/JobPreview.tsx]: This should be localized and/or conditionally shown based on the prop.

File: resources/assets/js/components/JobPreview.tsx [hunk 4]
```diff
@@ -68,67 +150,183 @@ const JobPreview: React.FunctionComponent<JobPreviewProps> = ({
...
+        </div>
+        <div data-c-grid-item="tp(1of2)">
+          <p data-c-colour="c1" data-c-margin="bottom(quarter)">
+            <FormattedMessage
+              id="jobPreview.overtime"
+              defaultMessage="Overtime"
+              description="Datum Label"
+            />
           </p>
+          <p>Not yet implemented.</p>
```
Comment (Khristophor) [inline resources/assets/js/components/JobPreview.tsx]: This should be localized and/or conditionally shown based on the prop.

File: resources/assets/js/components/JobDetails/JobDetails.tsx [hunk 5]
```diff
@@ -367,6 +368,7 @@ const jobToValues = (job: Job | null, locale: string): JobFormValues =>
         securityLevel: job.security_clearance_id || "",
         language: job.language_requirement_id || "",
         city: job[locale].city || "",
+        education: job[locale].education || "",
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetails.tsx]: Education isn't actually part of the form, so it doesn't really need to be set in JobToValues or UpdateJobWithValues

File: resources/assets/js/components/JobDetails/JobDetails.tsx [hunk 6]
```diff
@@ -950,7 +955,10 @@ const JobDetails: React.FunctionComponent<
                         province={intl.formatMessage(
                           provinceName(Number(values.province)),
                         )}
+                        education={values.education}
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetails.tsx]: Education should come from the Job object here

File: resources/assets/js/components/JobDetails/JobDetails.tsx [hunk 7]
```diff
@@ -950,7 +955,10 @@ const JobDetails: React.FunctionComponent<
...
                         termLength={Number(values.termLength)}
+                        telework={String(values.telework)}
```
Comment (Khristophor) [inline resources/assets/js/components/JobDetails/JobDetails.tsx]: I think it makes sense to pass the already localized versions of telework and flexhours here, rather than duplicating all the translations within the JobPreview component.

It also tightly couples the preview to the JobDetails form.

File: resources/assets/js/components/JobPreview.tsx [hunk 8]
```diff
@@ -11,36 +17,112 @@ export interface JobPreviewProps {
...
   classification: string;
   /** Level for the classification, i.e. 03 */
   level: string;
+  /** How often are Flex Hours allowed */
+  flexHours: string;
+  // Description of Required Education
+  education: string;
+  // Telework is allowed
+  telework: string;
+  // Travel opportunities;
```
Comment (Khristophor) [inline resources/assets/js/components/JobPreview.tsx]: Is this for another prop?

File: resources/assets/js/translations/locales/fr.json [hunk 9]
```diff
@@ -109,6 +108,24 @@
   "formValidation.tooLong": "Too long!",
   "formValidation.tooShort": "Too short!",
   "job.daysSinceClosed": "{dayCount, plural, =0 {Aucun jour} one {# jour} other {# jours}} depuis la fermeture",
+  "jobBuilder.culturePace.01.description": "Our deadlines are tight, we balance several tasks at the same time, and our priorities are always changing. Our work should come with running shoes!",
```
Comment (Khristophor) [inline resources/assets/js/translations/locales/fr.json]: Most of these look like translations for other files, they should be committed as part of another PR (or multiple PR's)

File: resources/assets/js/components/JobPreview.tsx (hunk 2 above)
Comment (xander-miller) [reply resources/assets/js/components/JobPreview.tsx]: Agreed I would like something that provides context to the header. how is? "Job Preview Card Info Label"

File: resources/assets/js/components/JobPreview.tsx (hunk 1 above)
Comment (xander-miller) [reply resources/assets/js/components/JobPreview.tsx]: Yeah, I was trying to split some of the differences between the old design and the new design. It was a stupid move in hind sight. Fixed. 👍

File: resources/assets/js/translations/locales/fr.json (hunk 9 above)
Comment (xander-miller) [reply resources/assets/js/translations/locales/fr.json]: That's a result of translations being added with out the `npm run translation` script being run. I will make a separate PR that is just updating the translations. That should fix this.

Comment (xander-miller) [reply resources/assets/js/translations/locales/fr.json]: Done. 👍

File: resources/assets/js/components/JobDetails/JobDetails.tsx (hunk 5 above)
Comment (xander-miller) [reply resources/assets/js/components/JobDetails/JobDetails.tsx]: What if the form is being used to update an existing job poster? I thought in that case the job information should already be there and passed through.

File: resources/assets/js/components/JobPreview.tsx (hunk 3 above)
Comment (xander-miller) [reply resources/assets/js/components/JobPreview.tsx]: Agreed, conditionally shown on the prop. +1

File: resources/assets/js/components/JobPreview.tsx (hunk 4 above)
Comment (xander-miller) [reply resources/assets/js/components/JobPreview.tsx]: Agreed, conditionally shown on the prop. 👍

File: resources/assets/js/components/JobDetails/JobDetails.tsx (hunk 5 above)
Comment (xander-miller) [reply resources/assets/js/components/JobDetails/JobDetails.tsx]: Yeah, now that I take a step back and think about it. It doesn't make sense. 👍

File: resources/assets/js/components/JobDetails/JobDetails.tsx (hunk 6 above)
Comment (xander-miller) [reply resources/assets/js/components/JobDetails/JobDetails.tsx]: Makes sense 👀 Done 👍

File: resources/assets/js/components/JobDetails/JobDetails.tsx (hunk 7 above)
Comment (xander-miller) [reply resources/assets/js/components/JobDetails/JobDetails.tsx]: Agreed. I'm using a different approach now. 👍

File: resources/assets/js/components/JobPreview.tsx (hunk 8 above)
Comment (xander-miller) [reply resources/assets/js/components/JobPreview.tsx]: Yes, and that prop is back. 👍
